hera_pspec/uvwindow.py

```diff
@@ -0,0 +1,1320 @@
+                    uvutils.polnum2str(pols[0], x_orientation=x_orientation))
+        else:
+            raise TypeError("Must feed pols as (tuple of) str or int.")
+        self.pols = pols
+
+        # spectral window
+        if spw_range is not None:
+            # checks on spw range input
+            spw_range = np.array(spw_range, dtype=int)
+            assert spw_range.size == 2, "spw_range must be fed as a tuple of "\
+                                        "frequency indices."
+            assert spw_range[1]-spw_range[0] > 0 and min(spw_range) >= 0,\
+                "Require non-zero spectral range."
+
+        # create list of FTBeam objects for each polarisation channel
+        self.ftbeam_obj_pol = []
+        # if not fed, creates appropriate FTBeam objects
+        if ftbeam_obj is None:
+            # create ftbeam_obj for first pol in the pair
+            self.ftbeam_obj_pol.append(FTBeam(self.pols[0], ftfile=ftfile, spw_range=spw_range,
+                                              verbose=verbose))
+            # copy object if polarisations in pair are identical
+            if self.pols[0] == self.pols[1]:
+                self.ftbeam_obj_pol.append(copy.deepcopy(self.ftbeam_obj_pol[0]))
+            # else, create new ftbeam object
+            else:
+                self.ftbeam_obj_pol.append(FTBeam(self.pols[1], ftfile=ftfile, spw_range=spw_range,
+                                                  verbose=verbose))
+        # if fed, read inputs
+        else:
+            ftbeam_obj = ftbeam_obj if isinstance(ftbeam_obj, (list, tuple, np.ndarray)) else [ftbeam_obj, ftbeam_obj]
+            # if input is only one object
+            if (ftbeam_obj[0] == ftbeam_obj[1]):
+                # check if polarisations in pair are identical
+                assert ftbeam_obj[0].pol == self.pols[0] and ftbeam_obj[1].pol == self.pols[1],\
+                    'If feeding only one ftbeam_obj, polarisations in pair must be identical'
+            # raise warning if ftfile was fed as it will be ignored
+            if ftfile is not None:
+                warnings.warn('Fed ftfile value will be overriden by input ftbeam_obj')
+                assert ftfile == ftbeam_obj[0].ft_file and ftfile == ftbeam_obj[1].ft_file,\
+                    "FT file of ftbeam_obj does not match ftfile of UVWindow"
+            # read object to fill self.ftbeam_obj_pol list
+            for ib, ftbeam in enumerate(ftbeam_obj):
+                ftbeam = copy.deepcopy(ftbeam)
+                # assert properties of ftbeam object fed are consistent with other inputs
+                assert ftbeam.check(), "There is an issue with the ft_beam object."
+                assert ftbeam.pol == self.pols[ib],\
+                    "Polarisation of ftbeam_obj does not match polarisation of UVWindow"
+                if spw_range is not None and ftbeam.spw_range != tuple(spw_range):
+                    ftbeam.update_spw(spw_range)
```

Comment:
I'd just leave it up to the user to set the spw_range they want on the FTBeam object. Don't let them pass in a potentially different one here.